microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -231,15 +232,46 @@ def _check_supported(self, node: Node) -> None:
         kdump = node.tools[KdumpBase]
         kdump.check_required_kernel_config()
 
+    def _get_resource_disk_dump_path(self, node: Node) -> str:
+        if node.shell.exists(
+            PurePosixPath("/var/log/cloud-init.log")
+        ) and node.shell.exists(PurePosixPath("/var/lib/cloud/instance")):
+            mount_point = "/mnt"
+        else:
+            mount_point = node.tools[Waagent].get_resource_disk_mount_point()
+
+        dump_path = mount_point + "/crash"
+        node.execute(
+            f"mkdir -p {dump_path}",
+            expected_exit_code=0,
+            expected_exit_code_failure_message=(f"Fail to create dir {dump_path}"),
+            shell=True,
+            sudo=True,
+        )
+        return dump_path
+
     def _kdump_test(self, node: Node, log_path: Path, log: Logger) -> None:
         try:
             self._check_supported(node)
         except UnsupportedDistroException as identifier:
             raise SkippedException(identifier)
 
         kdump = node.tools[KdumpBase]
+
+        memory_size = node.execute(
+            "free -h | grep Mem | awk '{print $2}'", shell=True, sudo=True
+        )
+        if "Ti" in memory_size.stdout and float(memory_size.stdout.strip("Ti")) > 1:
```

Comment:
Why not always use the resource disk if it exists?

Reply:
Yes, this is a good way. 

I tried to use another location to save the vmcore on Ubuntu and Suse. The dump is all failed. I was trying to figure out if it needs another configuration or if it has some issues when it changes to another location. After understanding the reason, we can change to using the resource disk by default.

I just added this way on Redhat, which can work well on Redhat.